src/NHibernate/Connection/ConnectionProvider.cs

```diff
@@ -24,6 +24,9 @@ public abstract partial class ConnectionProvider : IConnectionProvider
 		/// <param name="conn">The <see cref="DbConnection"/> to clean up.</param>
 		public virtual void CloseConnection(DbConnection conn)
 		{
+			if (conn == null)
+				throw new ArgumentNullException(nameof(conn));
```

Comment:
We can choose to just exit instead of throwing. But since the case was clearly not supported, causing a NRE in the try then a second NRE in the catch, I consider the current code base is never meant to supply `null` to this method, so it looks better to throw.